File: typescript/commands/ping.ts
import {Message} from "discord.js"
import {Discord} from "../structures/Discord.ts"

export default class PingCommand {
    private readonly discord: Discord
    private readonly message: Message

    public constructor(discord: Discord, message: Message) {
        this.discord = discord
        this.message = message
    }

    public run = async (args: string[]) => {
        const discord = this.discord
        const message = this.message

        return message.reply("Pong!")
    }
}

File: kotlin/commands/Ping.kt
package commands

import dev.kord.core.behavior.reply
import dev.kord.core.entity.Message
import structures.Discord
import structures.Command

class Ping(discord: Discord, message: Message) : Command(discord, message) {
    override suspend fun run(args: List<String>) {
        message.reply {
            content = "Pong!"
        }
    }
}